assets/admin/index.js

```diff
@@ -15,6 +15,7 @@ import 'sulu-route-bundle';
 import 'sulu-search-bundle';
 import 'sulu-security-bundle';
 import 'sulu-snippet-bundle';
+import 'sulu-trash-bundle';
```

Comment:
Actually the trash bundle should not be have custom JS as it should be able to be removed and so I would make a more general component which can live in the Admin Bundle for the restore then having custom JS inside the trash bundle.

Reply:
why should the bundle not include any javascript? the `RestoreItemAction` that uses the `RestoreFormOverlay` feels quite specific to me. i am not sure if its worth to extract a generalized abstraction for it

Reply:
The same why the article bundle should not include custom JS as the bundle should be optional, thats ehy I would prefer that it has no custom JS. What does the overlay do is it not just an action with a confirm dialog which is shown? Currently we create a lot of very specific item and toolbar action which are not really r usable in other cases.

Reply:
the article bundle is an external bundle that would force projects to create their own build, thats a big difference to me 🙂 
the item action displays a form overlay with a dynamic form key or a dialog with a message depending on the resource key of the clicked item. i am sure this could be abstracted in some way, but i dont think there are a lot of usecases where this could be reused

Reply:
> the article bundle is an external bundle

And we should more and more make it possible that the bundles should exist as its own packages, and the admin bundle should provide as much general components as possible. Specially in cases of toolbar actions and item action we should focus on providing more general reusable solutions. There where in the past sometimes discussions for some projects where sulu would just be used as DMS without the CMS part and it would be great to have as many UI components inside the Admin Bundle and not in external packages. I'm fine with let it currently as it is but I think we should focus more on creating reusable toolbar action and item action, which projects can reuse as creating again action components for specific usecases which are not reusable.